lib/network/kmsAWS/Client.ts

```diff
@@ -163,30 +135,23 @@ export default class Client implements KMSInterface {
             KeyId: masterKeyId,
             PendingWindowInDays: 7,
         };
-        this.client.scheduleKeyDeletion(params, (err: AWSError, data) => {
-            if (err) {
-                if (err.code === 'NotFoundException' || err.code === 'KMSInvalidStateException') {
-                    // master key does not exist or is already pending deletion
-                    logger.info('AWS KMS: key does not exist or is already pending deletion',
-                        { masterKeyId, error: err });
-                    cb(null);
-                    return;
-                }
-
-                const error = arsenalErrorAWSKMS(err);
-                logger.error('AWS KMS: failed to delete master encryption key', { err });
-                cb(error);
-                return;
-            }
-
+        this.client.send(new ScheduleKeyDeletionCommand(params)).then(data => {
             if (data?.KeyState && data.KeyState !== 'PendingDeletion') {
                 const error = arsenalErrorAWSKMS('key is not in PendingDeletion state');
-                logger.error('AWS KMS: failed to delete master encryption key', { err, data });
+                logger.error('AWS KMS: failed to delete master encryption key', { data });
                 cb(error);
                 return;
             }
-
             cb(null);
+        }).catch(err => {
+            if (err.name === 'NotFoundException' || err.name === 'KMSInvalidStateException') {
```

Comment:
nit: I think it's slightly nicer to use instanceof https://aws.amazon.com/blogs/developer/service-error-handling-modular-aws-sdk-js/

```suggestion
            if (err instanceof NotFoundException || err instanceof KMSInvalidStateException) {
```

Reply:
nicer and more importantly safer & recommended approach in SDK docs